src/pages/index.js

```diff
@@ -0,0 +1,359 @@
+      handleLike(evt, data._id, cardLikedBtn));
+  
+    cardImageEl.addEventListener("click", () => {
+      openModal(previewModal);
+      previewModalCaptionEl.textContent = data.name;
+      previewModalImageEl.src = data.link;
+      previewModalImageEl.alt = data.name;
+    });
+  
+    cardDeleteBtn.addEventListener("click", () => 
+      handleDeleteCard(cardElement, data._id, cardDeleteBtn));
+      return cardElement;
+  }
+  
+  function handleDeleteCard(cardElement, cardId, cardDeleteBtn) {
+    selectedCard = cardElement;
+    selectedCardId = cardId;
+    openModal(deleteModal);
+  }
+  
+  function handleDeleteSubmit(evt) {
+    evt.preventDefault();
+  
+    setButtonText(avatarDeletetBtn, true, "Delete", "Deleting...");
+  
+    api.deleteCard(selectedCardId)
+      .then(() => {
+        selectedCard.remove();
+        closeModal(deleteModal);
+      })
+      .catch(console.error)
+      .finally(() => {
+        setButtonText(avatarDeletetBtn, false, "Delete");
+      });
+  }
+  
+  function handleEditFormSubmit(evt) {
+    evt.preventDefault();
+    const profileSubmitBtn = evt.submitter;
+    setButtonText(profileSubmitBtn, true, "Save", "Saving...");
+  
+    const updatedUserInfo = {
+      name: editModalNameInput.value,
+      about: editModalAboutInput.value
+    };
+  
+    api.editUserInfo(updatedUserInfo)
+      .then((data) => {
+        profileName.textContent = data.name;
+        profileDescription.textContent = data.about;
+        closeModal(editModal);
+        editFormElement.reset();
+        disabledButton(profileSubmitBtn, settings); 
+      })
+      .catch((error) => {
+        console.error(error);
+      })
+      .finally(() => {
+        setButtonText(profileSubmitBtn, false, "Save");
+      });
+  }
+  
+  function toggleSubmitButton() {
+    const isInputValid = cardNameInput.value.trim() !== "" && cardLinkInput.value.trim() !== "";
+    disabledButton(cardSubmitBtn, !isInputValid); 
+  }
+  
+  function handleAddCardSubmit(evt) {
+    evt.preventDefault();
+    disabledButton(cardSubmitBtn, settings); 
+    setButtonText(cardSubmitBtn, true, "Save", "Saving...");
+  
+    const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  
+    api.createCard(inputValues)
+      .then((data) => {
+        const cardElement = getCardElement(data);
+        cardsList.prepend(cardElement);
+        closeModal(cardModal);
+        cardForm.reset();
+        disabledButton(cardSubmitBtn, settings);
+        cardSubmitBtn.classList.add("modal__submit-btn_disabled"); 
+      })
+      .catch((error) => {
+        console.error(error);
+      })
+      .finally(() => {
+        setButtonText(cardSubmitBtn, false, "Save"); 
+      });
+  }
+  
+  function handleAvatarSubmit(evt) {
+    evt.preventDefault();
+    setButtonText(avatarSubmitBtn, true, "Save", "Saving...");
+    avatarSubmitBtn.disabled = true;
+    avatarSubmitBtn.classList.add("modal__submit-btn_disabled");
```

Comment:
You don't need to disable the button before the request. It's ok if it's active here


But if you want to disable it, then you should reuse the `disabledButton` function


```js
      disabledButton(avatarSubmitBtn, settings);


```